std/debug/index.zig

```diff
@@ -71,6 +71,19 @@ pub fn getSelfDebugInfo() !*DebugInfo {
     }
 }
 
+pub fn supportsAnsi(file: os.File) bool {
```

Comment:
the problem is that `std.os.isTty` returns `true` for the windows console even though it does't support ansi escape codes, so i needed a different function.

Reply:
I see. I think this is still not quite what we want. This is testing if `GetConsoleMode` fails:
> If the function succeeds, the return value is nonzero.
> If the function fails, the return value is zero. To get extended error information, call GetLastError.

It's strange to conclude that a File supports ANSI because we get an error for getting the console mode.
I think instead we would want to return `true` for Windows if `windowsIsCygwinPty` return true.

I would also propose to put this function right next to `isTty` and accept a `FileHandle` (same as `isTty`) rather than a File. Maybe sometime in the future we refactor these both to use `File`; maybe we don't. But I think they should be together.

One more proposal: name it `supportsAnsiEscapeCodes`. Reasoning:
 *  descriptive; avoids confusion from people thinking a file would support ANSI character encodings or something like that.
 * unlikely to be called many times; a longer name is fine

Reply:
i was looking at `windowsIsCygwinPty` and i was wondering why we return `true` when `GetFileInformationByHandleEx` fails. when i change the return value to `false` there everything works as expected.

Reply:
Hmm that does seem like a mistake. I think we should change it to `false`.

Reply:
I think I got the initial implementation from some rust code: https://gist.github.com/andrewrk/182ace5dee6c4025d8c4b0ca22ca98ca
and I didn't validate this logic.

Reply:
This seems to confirm that we should change it to `false` like you said: https://github.com/ocaml/ocaml/pull/1406/files

Reply:
would you mind if i moved these functions into a `terminal` submodule under `os`?

Reply:
Go for it :+1:

Reply:
i'll move that to a separate pull request as i plan to add more functionality, like moving the cursor and querying terminal size.

Reply:
Sounds good. So this PR is ready then?

Reply:
yup.